rs/https_outcalls/adapter/src/rpc_server.rs

```diff
@@ -98,8 +114,148 @@ impl CanisterHttp {
         Self {
             client,
             socks_client,
+            cache: Arc::new(RwLock::new(BTreeMap::new())),
             logger,
             metrics: AdapterMetrics::new(metrics),
+            http_connect_timeout_secs: config.http_connect_timeout_secs,
+        }
+    }
+
+    fn create_socks_client_for_address(
+        &self,
+        address: &str,
+    ) -> Option<Client<HttpsConnector<SocksConnector<HttpConnector>>, OutboundRequestBody>> {
+        // Create a new HTTP connector
+        let mut http_connector = HttpConnector::new();
+        http_connector.enforce_http(false);
+        http_connector
+            .set_connect_timeout(Some(Duration::from_secs(self.http_connect_timeout_secs)));
+
+        match address.parse() {
+            Ok(proxy_addr) => {
+                let proxy_connector = SocksConnector {
+                    proxy_addr,
+                    auth: None,
+                    connector: http_connector,
+                };
+
+                let proxied_https_connector = HttpsConnectorBuilder::new()
+                    .with_native_roots()
+                    .expect("Failed to set native roots")
+                    .https_only()
+                    .enable_all_versions()
+                    .wrap_connector(proxy_connector);
+
+                let socks_client = Client::builder(TokioExecutor::new())
+                    .build::<_, Full<Bytes>>(proxied_https_connector);
+
+                Some(socks_client)
+            }
+            Err(e) => {
+                debug!(self.logger, "Failed to parse SOCKS address: {}", e);
+                None
+            }
+        }
+    }
+
+    fn compare_results(
+        &self,
+        result: &Result<http::Response<Incoming>, String>,
+        dl_result: &Result<http::Response<Incoming>, String>,
+    ) {
+        match (result, dl_result) {
+            (Ok(_), Ok(_)) => {
+                debug!(self.logger, "SOCKS_PROXY_DL: Both requests succeeded");
+            }
+            (Err(_), Err(_)) => {
+                debug!(self.logger, "SOCKS_PROXY_DL: Both requests failed");
+            }
+            (Ok(_), Err(err)) => {
+                debug!(
+                    self.logger,
+                    "SOCKS_PROXY_DL: regular request succeeded, DL request failed with error {}",
+                    err,
+                );
+            }
+            (Err(err), Ok(_)) => {
+                debug!(
+                    self.logger,
+                    "SOCKS_PROXY_DL: DL request succeeded, regular request failed with error {}",
+                    err,
+                );
+            }
+        }
+    }
+
+    async fn https_outcall_dl_socks_proxy(
```

Comment:
what does dl stand for ? :)

Reply:
Thanks I removed the DL. It stood for dark launch